orderer/consensus/etcdraft/chain.go

```diff
@@ -911,6 +925,45 @@ func (c *Chain) ordered(msg *orderer.SubmitRequest) (batches [][]*common.Envelop
 			}
 		}
 
+		if c.checkForEvictionNCertRotation(msg.Payload) {
+
+			if !atomic.CompareAndSwapUint32(&c.leadershipTransferInProgress, 0, 1) {
+				c.logger.Warnf("A reconfiguration transaction is already in progress, ignoring a subsequent transaction")
+				return
+			}
+
+			go func() {
+				defer atomic.StoreUint32(&c.leadershipTransferInProgress, 0)
+
+				for attempt := 1; attempt <= AbdicationMaxAttempts; attempt++ {
+					if err := c.Node.abdicateLeader(c.raftID); err != nil {
+						// If there is no leader, abort and do not retry.
+						// Return early to prevent re-submission of the transaction
+						if err == ErrNoLeader || err == ErrChainHalting {
+							return
+						}
```

Comment:
You have a point, but returning would not be a good decision, as it would result in the transaction being dropped. 
I changed the code and instead of returning `ErrNotALeader` we now return `nil`, and this causes the forwarding of the transaction to the leader.